src/main/java/dog.java

```diff
@@ -0,0 +1,40 @@
+public class dog implements Tradable, Domesticatable, Drivable {
```

Comment:
naming convention: usually we name the class with the first letter being capitalized?